src/java.base/share/classes/java/lang/invoke/InnerClassLambdaMetafactory.java

```diff
@@ -175,13 +175,15 @@ public InnerClassLambdaMetafactory(MethodHandles.Lookup caller,
         lambdaClassName = lambdaClassName(targetClass);
         // If the target class invokes a protected method inherited from a
         // superclass in a different package, or does 'invokespecial', the
-        // lambda class has no access to the resolved method. Instead, we need
-        // to pass the live implementation method handle to the proxy class
-        // to invoke directly. (javac prefers to avoid this situation by
-        // generating bridges in the target class)
+        // lambda class has no access to the resolved method, or does
+        // 'invokestatic' on a hidden class. Instead, we need to pass the
+        // live implementation method handle to the proxy class to invoke
+        // directly. (javac prefers to avoid this situation by generating
+        // bridges in the target class)
```

Comment:
```suggestion
        // lambda class has no access to the resolved method, or does
        // 'invokestatic' on a hidden class which cannot be resolved by name.
        // Instead, we need to pass the live implementation method handle to
        // the proxy class to invoke directly. (javac prefers to avoid this
        // situation by generating bridges in the target class)
```